client/actions/auth.js

```diff
@@ -60,37 +60,120 @@ export function logout(logoutUrl) {
   };
 }
 
+// calls checkSession to refresh idToken
+function refreshToken () {
+  return new Promise((resolve, reject) => {
+    // invoke check session to get a new token
+    webAuth.checkSession({},
+      function(err, result) {
+        if (err) { // there was an error
```

Comment:
we should be checking for a login_required error here and redirecting to authorize if we get one.